x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/request/amazonbedrock/AmazonBedrockJsonBuilder.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.request.amazonbedrock;
+
+import com.fasterxml.jackson.core.JsonFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+public class AmazonBedrockJsonBuilder {
+
+    private final AmazonBedrockJsonWriter jsonWriter;
+
+    public AmazonBedrockJsonBuilder(AmazonBedrockJsonWriter jsonWriter) {
+        this.jsonWriter = jsonWriter;
+    }
+
+    public String getStringContent() throws IOException {
+        var factory = new JsonFactory();
```

Comment:
For whatever weird reason - whenever I convert XContent to a string, and try to use that to send to the InvokeModelRequest body - Amazon gives me a "Bad Request"... using the Jackson json generator directly here does not... 🤷  - it provided for hours of frustration, and I never did get a clear answer why XContent wouldn't work...

Reply:
Oh interesting. How about we create an issue to investigate this more later? Just tag ML and we can add it to our backlog. I think what you have is fine, here's some info for the issue:

In the issue could we add a reference to the `JsonFactory` docs: https://fasterxml.github.io/jackson-core/javadoc/2.6/com/fasterxml/jackson/core/JsonFactory.html and include some information like this:

If we end up being unable to leverage XContent it might be worth investigating if it's beneficial to only instantiate `JsonFactory` once. The docs state:

```
Typically applications and services use only a single globally shared factory instance, unless they need differently configured factories
```

But then it also states:

```
Creation of a factory instance is a light-weight operation, and since there is no need for pluggable alternative implementations (as there is no "standard" JSON processor API to implement), the default constructor is used for constructing factory instances.
```

🤷‍♂️

Reply:
> If we end up being unable to leverage XContent it might be worth investigating if it's beneficial to only instantiate JsonFactory once.

Ah - great point... I'll do a bit more investigation this morning (I'd rather use XContent) - and if I cannot get it to work, I'll try a single `JsonFactory` static instance..

Reply:
Good news 🎉  - no idea what happened in my earlier code, but I've re-added in the ToXContent() JSON generation and it's working now (with the latest commit)... I probably had something upstream earlier that was broken, but 🤷  - this is much cleaner, and less prone to external issues :)

Reply:
Nice!